sdk/src/Core/Amazon.Runtime/Pipeline/ErrorHandler/ErrorHandler.cs

```diff
@@ -211,5 +211,48 @@ private bool ProcessException(IExecutionContext executionContext, Exception exce
             // No match found, rethrow the original exception.
             return true;
         }
+
+#if AWS_ASYNC_API
+        /// <summary>
+        /// Processes an exception by invoking a matching exception handler
+        /// for the given exception.
+        /// </summary>
+        /// <param name="executionContext">The execution context, it contains the
+        /// request and response context.</param>
+        /// <param name="exception">The exception to be processed.</param>
+        /// <returns>
+        /// This method returns a boolean value which indicates if the original exception
+        /// should be rethrown.
+        /// This method can also throw a new exception that may be thrown by exception
+        /// processing by a matching exception handler.
+        /// </returns>
+        private async System.Threading.Tasks.Task<bool> ProcessExceptionAsync(IExecutionContext executionContext, Exception exception)
+        {
+            // Log the exception
+            this.Logger.Error(exception, "An exception of type {0} was handled in ErrorHandler.", exception.GetType().Name);
```

Comment:
We need to refactor this and ProcessException to not have so much duplicate code.